src/pass/llvmgen.cpp

```diff
@@ -377,7 +378,10 @@ void LLVMGen::visit(Stmts& stmts)
 
 Value* LLVMGen::visit(Alloc& alloc)
 {
-    return builder()->CreateAlloca(lltype(alloc.type), eval(alloc.size));
+    // 5 for local address space
+    // see https://llvm.org/docs/NVPTXUsage.html#address-spaces
+    auto type = PointerType::get(lltype(alloc.type.dtypes[0]), 5U);
+    return builder()->CreateAlloca(type, eval(alloc.size));
```

Comment:
this is not generalizable. this visitor is supposed to work for any allocation type. but here you are only allocating a pointer type of lltype(alloc.type.dtypes[0]).